.github/workflows/build-image-pr.yml

```
@@ -1,4 +1,4 @@
-name: Test controller image build
+name: Test controller image build and deployment
 
 on:
   pull_request_target:
```

Is there a reason behind using `pull_request_target`, otherwise I think we can easily change this to `pull_request` such that the PR check is actually testing the content in the PR, wdyt? (furthermore you can also check this new workflow here in this PR)

```suggestion
  pull_request:
```